app_dart/test/request_handlers/check_flaky_tests_and_update_github_test.dart

```diff
@@ -0,0 +1,447 @@
+      expect(issueRequest.assignee, expectedSemanticsIntegrationTestResponseAssignee);
+      expect(const ListEquality<String>().equals(issueRequest.labels, expectedSemanticsIntegrationTestResponseLabels),
+          isTrue);
+
+      // Verify tree is created correctly.
+      captured = verify(mockGitService.createTree(captureAny, captureAny)).captured;
+      expect(captured.length, 2);
+      expect(captured[0].toString(), '$kCurrentUserLogin/flutter');
+      expect(captured[1], isA<CreateGitTree>());
+      final CreateGitTree tree = captured[1] as CreateGitTree;
+      expect(tree.baseTree, kCurrentMasterSHA);
+      expect(tree.entries.length, 1);
+      expect(tree.entries[0].content, expectedSemanticsIntegrationTestCiYamlContent);
+      expect(tree.entries[0].path, CheckForFlakyTestAndUpdateGithub.kCiYamlPath);
+      expect(tree.entries[0].mode, CheckForFlakyTestAndUpdateGithub.kModifyMode);
+      expect(tree.entries[0].type, CheckForFlakyTestAndUpdateGithub.kModifyType);
+
+      // Verify commit is created correctly.
+      captured = verify(mockGitService.createCommit(captureAny, captureAny)).captured;
+      expect(captured.length, 2);
+      expect(captured[0].toString(), '$kCurrentUserLogin/flutter');
+      expect(captured[1], isA<CreateGitCommit>());
+      final CreateGitCommit commit = captured[1] as CreateGitCommit;
+      expect(commit.message, expectedSemanticsIntegrationTestPullRequestTitle);
+      expect(commit.author.name, kCurrentUserName);
+      expect(commit.author.email, kCurrentUserEmail);
+      expect(commit.committer.name, kCurrentUserName);
+      expect(commit.committer.email, kCurrentUserEmail);
+      expect(commit.tree, expectedSemanticsIntegrationTestTreeSha);
+      expect(commit.parents.length, 1);
+      expect(commit.parents[0], kCurrentMasterSHA);
+
+      // Verify reference is created correctly.
+      captured = verify(mockGitService.createReference(captureAny, captureAny, captureAny)).captured;
+      expect(captured.length, 3);
+      expect(captured[0].toString(), '$kCurrentUserLogin/flutter');
+      expect(captured[2], expectedSemanticsIntegrationTestTreeSha);
+      final String ref = captured[1] as String;
+
+      // Verify pr is created correctly.
+      captured = verify(mockPullRequestsService.create(captureAny, captureAny)).captured;
+      expect(captured.length, 2);
+      expect(captured[0].toString(), config.flutterSlug.toString());
+      expect(captured[1], isA<CreatePullRequest>());
+      final CreatePullRequest pr = captured[1] as CreatePullRequest;
+      expect(pr.title, expectedSemanticsIntegrationTestPullRequestTitle);
+      expect(pr.body, expectedSemanticsIntegrationTestPullRequestBody);
+      expect(pr.head, '$kCurrentUserLogin:$ref');
+      expect(pr.base, 'refs/${CheckForFlakyTestAndUpdateGithub.kMasterRefs}');
+
+      expect(result['Statuses'], 'success');
+    });
+
+    test('Can file issue but not pr for framework host-only test', () async {
```

Comment:
I guess a pr should also be filed for framework host only tests. The only exception is the shard tests, which contains multiple sub unit tests.